prov/tcp/src/tcpx_progress.c

```diff
@@ -255,19 +313,41 @@ static int tcpx_get_rx_pe_entry(struct tcpx_rx_detect *rx_detect,
 		pe_entry->flags = TCPX_NO_COMPLETION;
 		pe_entry->done_len = sizeof(rx_detect->hdr);
 
-		ret = tcpx_process_rx_rma_data(pe_entry);
-		if (ret)
+		ret = tcpx_validate_rx_rma_data(pe_entry, FI_REMOTE_WRITE);
+		if (ret) {
+			FI_WARN(&tcpx_prov, FI_LOG_DOMAIN,
+				"invalid rma data\n");
+			tcpx_pe_entry_release(tcpx_cq, pe_entry);
 			return ret;
+		}
 
+		tcpx_copy_rma_iov_to_msg_iov(pe_entry);
 		break;
-	case ofi_op_read_req:
 	case ofi_op_read_rsp:
-	case ofi_op_write_rsp:
-		/* todo  complete these cases for rma*/
+		if (dlist_empty(&tcpx_ep->rma_list.list))
+			return -FI_EINVAL;
```

Comment:
This check shouldn't be needed, since dlist_find_first_match() will fail anyway, plus the list should really never be empty